datadog/data_source_datadog_ip_ranges.go

```diff
@@ -154,37 +197,25 @@ func dataSourceDatadogIPRangesRead(ctx context.Context, d *schema.ResourceData,
 	synthetics := ipAddressesPtr.GetSynthetics()
 	webhook := ipAddressesPtr.GetWebhooks()
 
-	if len(agents.GetPrefixesIpv4())+len(api.GetPrefixesIpv4())+
-		len(apm.GetPrefixesIpv4())+len(logs.GetPrefixesIpv4())+
-		len(process.GetPrefixesIpv4())+len(synthetics.GetPrefixesIpv4())+
-		len(webhook.GetPrefixesIpv4())+len(agents.GetPrefixesIpv6())+
-		len(api.GetPrefixesIpv6())+len(apm.GetPrefixesIpv6())+
-		len(logs.GetPrefixesIpv6())+len(process.GetPrefixesIpv6())+
-		len(synthetics.GetPrefixesIpv6())+len(webhook.GetPrefixesIpv6())+
-		len(synthetics.GetPrefixesIpv4ByLocation())+len(synthetics.GetPrefixesIpv6ByLocation()) > 0 {
-		d.SetId("datadog-ip-ranges")
-	}
-
-	// Set ranges when the list is not empty
+	// Set model values from response
 	// v4
-	d.Set("agents_ipv4", agents.GetPrefixesIpv4())
-	d.Set("api_ipv4", api.GetPrefixesIpv4())
-	d.Set("apm_ipv4", apm.GetPrefixesIpv4())
-	d.Set("logs_ipv4", logs.GetPrefixesIpv4())
-	d.Set("orchestrator_ipv4", orchestrator.GetPrefixesIpv4())
-	d.Set("process_ipv4", process.GetPrefixesIpv4())
-	d.Set("synthetics_ipv4", synthetics.GetPrefixesIpv4())
-	d.Set("webhooks_ipv4", webhook.GetPrefixesIpv4())
-
+	state.AgentsIpv4, _ = types.ListValueFrom(ctx, types.StringType, agents.GetPrefixesIpv4())
+	state.APIIpv4, _ = types.ListValueFrom(ctx, types.StringType, api.GetPrefixesIpv4())
+	state.APMIpv4, _ = types.ListValueFrom(ctx, types.StringType, apm.GetPrefixesIpv4())
+	state.LogsIpv4, _ = types.ListValueFrom(ctx, types.StringType, logs.GetPrefixesIpv4())
+	state.OrchestratorIpv4, _ = types.ListValueFrom(ctx, types.StringType, orchestrator.GetPrefixesIpv4())
+	state.ProcessIpv4, _ = types.ListValueFrom(ctx, types.StringType, process.GetPrefixesIpv4())
+	state.SyntheticsIpv4, _ = types.ListValueFrom(ctx, types.StringType, synthetics.GetPrefixesIpv4())
+	state.WebhooksIpv4, _ = types.ListValueFrom(ctx, types.StringType, webhook.GetPrefixesIpv4())
 	// v6
-	d.Set("agents_ipv6", agents.GetPrefixesIpv6())
-	d.Set("api_ipv6", api.GetPrefixesIpv6())
-	d.Set("apm_ipv6", apm.GetPrefixesIpv6())
-	d.Set("logs_ipv6", logs.GetPrefixesIpv6())
-	d.Set("orchestrator_ipv6", orchestrator.GetPrefixesIpv6())
-	d.Set("process_ipv6", process.GetPrefixesIpv6())
-	d.Set("synthetics_ipv6", synthetics.GetPrefixesIpv6())
-	d.Set("webhooks_ipv6", webhook.GetPrefixesIpv6())
+	state.AgentsIpv6, _ = types.ListValueFrom(ctx, types.StringType, agents.GetPrefixesIpv4())
```

Comment:
Should we care about the diag returned here? I remember errors from Set calls being useful sometimes.